common/fs/src/cache/mod.rs

```diff
@@ -260,14 +260,12 @@ impl FileSystem {
 
     pub fn stream_events(
         fs: Arc<Mutex<FileSystem>>,
-        buf: &mut [u8],
-    ) -> Result<impl Stream<Item = (Result<Event, Error>, EventTimestamp)> + '_, std::io::Error>
-    {
+    ) -> Result<impl Stream<Item = (Result<Event, Error>, EventTimestamp)>, std::io::Error> {
```

Comment:
Dropping the mutable buffer threaded through this code is required for the async blocks.